Stripe/STPCustomerContext.swift

```diff
@@ -0,0 +1,372 @@
+    self.init(keyProvider: keyProvider, apiClient: STPAPIClient.shared)
+  }
+
+  /// Initializes a new `STPCustomerContext` with the specified key provider.
+  /// Upon initialization, a CustomerContext will fetch a new ephemeral key from
+  /// your backend and use it to prefetch the customer object specified in the key.
+  /// Subsequent customer and payment method retrievals (e.g. by `STPPaymentContext`)
+  /// will return the prefetched customer / attached payment methods immediately if
+  /// its age does not exceed 60 seconds.
+  /// - Parameters:
+  ///   - keyProvider:   The key provider the customer context will use.
+  ///   - apiClient:       The API Client to use to make requests.
+  /// - Returns: the newly-instantiated customer context.
+  @objc(initWithKeyProvider:apiClient:)
+  public convenience init(
+    keyProvider: STPCustomerEphemeralKeyProvider?, apiClient: STPAPIClient
+  ) {
+    let keyManager = STPEphemeralKeyManager(
+      keyProvider: keyProvider,
+      apiVersion: STPAPIClient.apiVersion,
+      performsEagerFetching: true)
+    self.init(keyManager: keyManager, apiClient: apiClient)
+  }
+
+  /// `STPCustomerContext` will cache its customer object and associated payment methods
+  /// for up to 60 seconds. If your current user logs out of your app and a new user logs
+  /// in, be sure to either call this method or create a new instance of `STPCustomerContext`.
+  /// On your backend, be sure to create and return a new ephemeral key for the
+  /// customer object associated with the new user.
+  @objc
+  public func clearCache() {
+    clearCachedCustomer()
+    clearCachedPaymentMethods()
+  }
+
+  private var _includeApplePayPaymentMethods = false
+  /// By default, `STPCustomerContext` will filter Apple Pay when it retrieves
+  /// Payment Methods. Apple Pay payment methods should generally not be re-used and
+  /// shouldn't be offered to customers as a new payment method (Apple Pay payment
+  /// methods may only be re-used for subscriptions).
+  /// If you are using `STPCustomerContext` to back your own UI and would like to
+  /// disable Apple Pay filtering, set this property to YES.
+  /// Note: If you are using `STPPaymentContext`, you should not change this property.
+  @objc public var includeApplePayPaymentMethods: Bool {
+    get {
+      _includeApplePayPaymentMethods
+    }
+    set(includeApplePayMethods) {
+      _includeApplePayPaymentMethods = includeApplePayMethods
+      customer?.updateSources(filteringApplePay: !includeApplePayMethods)
+    }
+  }
+
+  private var _customer: STPCustomer?
+  private var customer: STPCustomer? {
+    get {
+      _customer
+    }
+    set(customer) {
+      _customer = customer
+      customerRetrievedDate = (customer) != nil ? Date() : nil
+    }
+  }
+  @objc internal var customerRetrievedDate: Date?
+
+  private var _paymentMethods: [STPPaymentMethod]?
+  private var paymentMethods: [STPPaymentMethod]? {
```

Comment:
some weird optionals here but not high-priority

Reply:
This is a cache, so being `nil` is valid and handled accordingly (though I think we'd write all this differently in Swift)